src/solver/conv_ck_igemm_fwd_v4r4r4_xdlops_nhwc.cpp

```diff
@@ -0,0 +1,258 @@
+        [config.ck_tunable_list_id];
+}
+
+} // namespace ck_utility
+
+bool PerformanceConvCkIgemmFwdV4r4r4XdlopsNhwc::SetNextValue(const ConvolutionContext&)
+{
+    if(ck_tunable_list_id <
+       ck::driver::ConvIgemmFwdV4r4r4XdlopsNhwcKyxcNhwk::GetTunableList().size() - 1)
+    {
+        ck_tunable_list_id++;
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+bool PerformanceConvCkIgemmFwdV4r4r4XdlopsNhwc::IsValid(const ConvolutionContext& ctx) const
+{
+    auto compile_param = ck::driver::CompileParameterConvIgemmFwdV4r4r4XdlopsNhwcKyxcNhwk{};
+    bool found         = false;
+
+    std::tie(compile_param, found) =
+        ck::driver::ConvIgemmFwdV4r4r4XdlopsNhwcKyxcNhwk::CalculateCompileParameterBasedOnTunable(
+            ck_utility::get_ck_convolution_problem_descriptor(ctx),
+            ck_utility::get_ck_tunable_conv_igemm_fwd_v4r4r4_xdlops_nhwc_kyxc_nhwk(*this));
+
+    if(!found)
+        return false;
+
+    return ck::driver::ConvIgemmFwdV4r4r4XdlopsNhwcKyxcNhwk::IsValidCompileParameter(
+        ck_utility::get_ck_convolution_problem_descriptor(ctx), compile_param);
+}
+
+bool ConvCkIgemmFwdV4r4r4XdlopsNhwc::IsApplicable(const ConvolutionContext& ctx) const
+{
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_CK_IGEMM_FWD_V4R4R4_XDLOPS_NHWC{}))
+        return false;
+    if(!ctx.use_hip_kernels)
+        return false;
+    if(!ck_utility::is_ck_supported_hardware(ctx.GetStream()))
+        return false;
+    if(!ctx.IsLayoutNHWC())
+        return false;
+    if(!ctx.direction.IsForward())
+        return false;
+    if(!ctx.Is2d())
+        return false;
+    if(!(ctx.IsFp32() or ctx.IsFp16()))
+        return false;
+    if(ctx.group_counts != 1)
+        return false;
+
+    {
+        // this kernel use int32_t for memory offset, which covers 2GB of memory maximum
+        const std::size_t max_index_range = std::size_t(2) * 1024 * 1024 * 1024;
+
+        if(!(ctx.bot_sz < max_index_range && ctx.weights_sz < max_index_range &&
+             ctx.top_sz < max_index_range))
+            return false;
+    }
+
+    return ck::driver::ConvIgemmFwdV4r4r4XdlopsNhwcKyxcNhwk::IsApplicable(
+        ck_utility::get_ck_convolution_problem_descriptor(ctx));
+}
+
+PerformanceConvCkIgemmFwdV4r4r4XdlopsNhwc
+ConvCkIgemmFwdV4r4r4XdlopsNhwc::GetPerformanceConfig(const ConvolutionContext& ctx) const
+{
+    for(int i = 0; i < ck::driver::ConvIgemmFwdV4r4r4XdlopsNhwcKyxcNhwk::GetTunableList().size();
+        ++i)
+    {
+        if(IsValidPerformanceConfig(ctx, i))
+        {
+            return {i};
+        }
+    }
+
+    MIOPEN_LOG_E("cannot find a valid performance config");
+
+    return {-1};
+}
+
+bool ConvCkIgemmFwdV4r4r4XdlopsNhwc::IsValidPerformanceConfig(
+    const ConvolutionContext& ctx, const PerformanceConvCkIgemmFwdV4r4r4XdlopsNhwc& config) const
+{
+    return config.IsValid(ctx);
+}
+
+ConvSolution
+ConvCkIgemmFwdV4r4r4XdlopsNhwc::GetSolution(const ConvolutionContext& ctx,
```

Comment:
:red_circle: The new solver is applicable for gfx90a and FP16. Please take care about this. Either provide FP16 ALT implementations in kernels, or disable when necessary.

Reply:
Pls let me know if you need more assistance on this.

Reply:
pls help me by checking commit "e30aa9e"

Reply:
Looks good!